File: CSharp-Two/StringsAndTextProcessing/ParseURL/ParseURL.cs
//### Problem 12. Parse URL
//*	Write a program that parses an URL address given in the format: 
// `[protocol]://[server]/[resource]` and extracts from it the `[protocol]`, `[server]` and `[resource]` elements.

//_Example:_ 

//|                 URL                  |                                     Information                                     |
//|:------------------------------------:|:----------------------------------------------------------------------------------------:|
//| `http://telerikacademy.com/Courses/Courses/Details/212` |
// [protocol] = `http` <br> [server] = `telerikacademy.com` <br> [resource] = `/Courses/Courses/Details/212` |

using System;
using System.Text;

class ParseURL
{
    static void Main()
    {
        Console.Write("Enter URL: ");
        string[] url = Console.ReadLine().Split(new string[] {"://", "/"}, StringSplitOptions.None );

        string protocol = url[0];
        string server = url[1];
        StringBuilder resource = new StringBuilder();
        resource.Append('/');
        for (int i = 2; i < url.Length; i++)
        {
            resource.Append(url[i]);
            resource.Append('/');
        }
        Console.WriteLine("Protocol: " + protocol);
        Console.WriteLine("Server: " + server);
        Console.WriteLine("Resource: " + resource);
    }
}

File: Javascript-Basics/Strings/ParseURL.js
//### Problem 7. Parse URL
//*	Write a script that parses an URL address given in the format: `[protocol]://[server]/[resource]`
//and extracts from it the `[protocol]`, `[server]` and `[resource]` elements.
//*	Return the elements in a JSON object.
//
//    _Example:_
//
//|                 URL                  |                                     result                                     |
//|:------------------------------------:|:----------------------------------------------------------------------------------------:|
//| `http://telerikacademy.com/Courses/Courses/Details/239` |  { protocol: `http`, <br> server: `telerikacademy.com` <br> resource: `/Courses/Courses/Details/239` |

var url1 = 'http://telerikacademy.com/Courses/ Courses/Details/239';
var url = 'ftp: // gosho . bg /Courses/Courses/ Details/239';

function getUrl(url) {
    var i, index, protocol, array, server, resourses;
    for (i = 0; i < url.length; i++) {
        url = url.replace(' ', '');
    }
    index = url.indexOf('//');
    protocol = url.substring(0, index - 1);
    url = url.split("//").pop();//delete from start of string until string''//'including,
    array = url.split('/');
    server = array[0];
    resourses = url.replace(server, '');
    console.log('Protocol: ' + protocol + '\n' + ' Server: ' + server + '\n' + ' Resourses: ' + resourses);
}
getUrl(url);
getUrl(url1);